src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java

```diff
@@ -0,0 +1,245 @@
+package org.fogbowcloud.arrebol.execution.docker;
+
+import static java.lang.Thread.sleep;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.log4j.Logger;
+import org.fogbowcloud.arrebol.execution.TaskExecutionResult;
+import org.fogbowcloud.arrebol.execution.TaskExecutor;
+import org.fogbowcloud.arrebol.execution.docker.exceptions.DockerRemoveContainerException;
+import org.fogbowcloud.arrebol.execution.docker.request.WorkerDockerRequestHelper;
+import org.fogbowcloud.arrebol.models.command.Command;
+import org.fogbowcloud.arrebol.models.command.CommandState;
+import org.fogbowcloud.arrebol.models.task.Task;
+
+/**
+ * This implementation of {@link TaskExecutor} manages the execution of a {@link Task} in a, possible remote,
+ * DockerContainer. A new container is created to execute every {@link Task} and destroy after the execution
+ * has finished. A task representation is sent to the container, which drives the execution of the
+ * commands itself. This objects monitors the execution until it ends on success or failure. If any container
+ * initialization error occurs, the return {@link TaskExecutionResult} indicates the Failure.
+ */
+public class DockerTaskExecutor implements TaskExecutor {
+
+    private static final long poolingPeriodTimeMs = 2000;
+    private static final String taskScriptFilePathPattern = "/tmp/%s.ts";
+    private static final String ecFilePathPattern = "/tmp/%s.ts.ec";
+
+    private final Logger LOGGER = Logger.getLogger(DockerTaskExecutor.class);
+
+    private WorkerDockerRequestHelper workerDockerRequestHelper;
+    private DockerExecutorHelper dockerExecutorHelper;
+
+    /**
+     * @param containerName Sets the name of the container, is an identifier.
+     * @param address Defines the address where requests for the Docker API should be made
+     * @param taskScriptContent Script that will be sent to the container with the function to
+     * execute the commands
+     * @param defaultImageId Image docker used as default if no one is specified in the task.
+     */
+    public DockerTaskExecutor(String containerName, String address, String taskScriptContent,
```

Comment:
maybe, we should not receive containerName, address and defaultImageId as args. Instead, receive the requestHelper directly. If you do so, I think we can delete on `set` method included for tests purpose.

Reply:
we can also receive the executorhelper as an args. It seems it would make our like easier

Reply:
Done.